.github/workflows/nightly-test-integrations-1.15.x.yml

```diff
@@ -212,7 +218,17 @@ jobs:
         consul-version: ["1.14", "1.15"]
     env:
       CONSUL_LATEST_VERSION: ${{ matrix.consul-version }}
-      ENVOY_VERSION: "1.24.6"
+      # ENVOY_VERSION should be the latest version supported by _all_ Consul versions in the
+      # matrix.consul-version, since we are testing upgrade from an older Consul version.
+      # In practice, this should be the highest Envoy version supported by the lowest non-LTS
+      # Consul version in the matrix (LTS versions receive additional Envoy version support).
+      #
+      # This value should be kept current in new nightly test workflows, and updated any time
+      # a new major Envoy release is added to the set supported by Consul versions in
+      # matrix.consul-version (i.e. whenever the highest common Envoy version across active
+      # Consul versions changes). The minor Envoy version does not necessarily need to be
+      # kept current for the purpose of these tests, but the major (1.N) version should be.
+      ENVOY_VERSION: "1.24.12"
```

Comment:
The edge case created by this argument requires either maintaining an `ENVOY_VERSIONS` file in inactive CE branches (so we can source the lowest common version for upgrade tests), or hard-coding it separate from the `ENVOY_VERSIONS` file.

After discussing w/ @DanStough , given this is going to be updated less frequently (typically just during Consul major releases in alignment w/ new nightly test files), we've decided not to try to solve for automatically calculating this value and add a note to the release guide instead.